src/main/java/edu/ucsb/cs156/happiercows/models/CreateCommonsParams.java

```diff
@@ -1,25 +1,31 @@
 package edu.ucsb.cs156.happiercows.models;
 
-import lombok.Data;
+import java.util.Collection;
+
+import lombok.AccessLevel;
 import lombok.AllArgsConstructor;
-import lombok.NoArgsConstructor;
 import lombok.Builder;
-import lombok.AccessLevel;
-
+import lombok.Data;
+import lombok.NoArgsConstructor;
 
 import org.springframework.security.core.GrantedAuthority;
 
 import edu.ucsb.cs156.happiercows.entities.User;
 
-import java.util.Collection;
-
 @Data
 @AllArgsConstructor
 @NoArgsConstructor(access = AccessLevel.PROTECTED)
 @Builder
-public class CreateCommonsParams {
+public class CreateCommonsParams
+{
   private String name;
-  private double cowPrice;
-  private double milkPrice;
-  private double startingBalance;
+  private String cowPrice;
+  private String milkPrice;
+  private String startingBalance;
```

Comment:
Should these be numeric values instead of strings?

Reply:
They have type `String` because of the `@RequestBody` annotation. My first try resulted in a type error. I thought Spring would coerce them like it does for query parameters but I guess not.

```java
@ApiParam("request body") @RequestBody CreateCommonsParams params
```

Reply:
Interesting, what types did you try to use? RequestBody should use the same HttpMessageConverter used by other parts of Spring, so I'm surprised that it's not supported.

Reply:
I may have used primitives.

Reply:
```
[ERROR] /Users/jackson/Documents/ucsb/cmpsc-156/team04-w22-6pm-HappyCows/src/test/java/edu/ucsb/cs156/happiercows/controllers/CommonsControllerTests.java:[67,21] incompatible types: java.lang.String cannot be converted to java.time.LocalDateTime
[ERROR] /Users/jackson/Documents/ucsb/cmpsc-156/team04-w22-6pm-HappyCows/src/test/java/edu/ucsb/cs156/happiercows/controllers/CommonsControllerTests.java:[72,17] incompatible types: java.lang.String cannot be converted to java.lang.Double
[ERROR] /Users/jackson/Documents/ucsb/cmpsc-156/team04-w22-6pm-HappyCows/src/test/java/edu/ucsb/cs156/happiercows/controllers/CommonsControllerTests.java:[102,17] incompatible types: java.lang.String cannot be converted to java.lang.Double
```

Reply:
You may need [`@DateTimeFormat`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/format/annotation/DateTimeFormat.html) and [`@NumberFormat`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/format/annotation/NumberFormat.html) to get those to work?

Reply:
Oh my gosh Bryan thank you! My only concern now is that the `Commons` and `CreateCommonsParams` types only differ by an `id` field and their respective annotations.

Reply:
I think it's natural to be concerned about code duplication, but there's a case to be made about decoupling the database internals from the public-facing API. For simple use cases, the model and entity will usually look very similar, that may change as new fields and relations are added. Having a separate model and entity also allows us to freely make changes to database persistence without requiring any changes on the frontend.

You might find better resources about this design pattern if you try reading up on Data Transfer Objects (DTOs). There are libraries that make this model/DTO <-> entity conversion much simpler, though we don't currently have that set up in these projects.